core/shared/src/test/scala/cats/parse/ParserTest.scala

```diff
@@ -668,6 +668,18 @@ class ParserTest extends munit.ScalaCheckSuite {
     parseTest(Parser.stringIn(List("foo", "foobar", "foofoo", "foobat")).string, "foobal", "foo")
   }
 
```

Comment:
please add generators that create any new elements to the AST (in this PR that would be `input`, but in a follow up it might be `WithCaret` or something). That way we can check all the other properties in the presence of this new node.

Reply:
I still want to address this to make sure all the previous laws pass.

see for instance:

https://github.com/typelevel/cats-parse/blob/8c9a27143abc30bc7f8000feb25d75e9a3a10906/core/shared/src/test/scala/cats/parse/ParserTest.scala#L167

as an example of how we want to transform some generators to make `withSpan` and `spanOf` and add them to here:

https://github.com/typelevel/cats-parse/blob/8c9a27143abc30bc7f8000feb25d75e9a3a10906/core/shared/src/test/scala/cats/parse/ParserTest.scala#L527

and:

https://github.com/typelevel/cats-parse/blob/8c9a27143abc30bc7f8000feb25d75e9a3a10906/core/shared/src/test/scala/cats/parse/ParserTest.scala#L555

Reply:
I see that you added the generators, but I don't see that they were added to the root generator so we don't exercise all the existing laws in the presence of these operations.